src/content/app/tools/blast/views/blast-form/BlastForm.tsx

```diff
@@ -60,12 +62,35 @@ const BlastForm = () => {
 
 const Main = () => {
   const isSmallViewport = useMediaQuery('(max-width: 1900px)');
+  const [sequencesValidity, setSequencesValidity] = useState<boolean[]>([]);
+  const { sequences } = useBlastForm();
+
+  const updateSequenceValidity = (index: number | null, status: boolean) => {
+    const array = [...sequencesValidity];
+    index = index === null ? sequences.length : index;
+    array[index] = status;
+    setSequencesValidity(array);
+  };
+
+  const removeSequenceValidity = (index: number) => {
+    setSequencesValidity(sequencesValidity.filter((value, i) => i !== index));
+  };
 
   if (isSmallViewport === null) {
     return null;
   }
 
-  return isSmallViewport ? <MainSmall /> : <MainLarge />;
+  return (
+    <BlastFormContext.Provider
+      value={{
+        updateSequenceValidity,
+        removeSequenceValidity,
+        sequencesValidity
+      }}
+    >
+      {isSmallViewport ? <MainSmall /> : <MainLarge />}
```

Comment:
By doing so, `MainSmall` and `MainLarge` will probably re-render every time the state of the `Main` container will update, which is probably going to be very frequently. We can avoid that by either wrapping those components in React.memo, or moving the logic related to the context into its own component, which would do the following:

```
type Props = {
  children: ReactNode
}

const BlastFormContextContainer = (props: Props) => {
   // create state
   // have function for updating the state 

  return (
    <BlastFormContext.Provider
      value={...}
    >{props.children}</BlastFormContext.Provider>
  );
}
```
and will be used in Main as:

```
<BlastFormContextContainer>{ isSmallViewport ? <MainSmall /> : <MainLarge /> }</BlastFormContextContainer>
```

Reply:
Done